### PR TITLE
test: Refactor flood tests into a separate test file

### PR DESCRIPTION
**Pull request by yanavlasov**
Commit Message:
Refactor flood tests into a separate test file

Additional Description: 
Moving flood related test into a separate file in preparation of adding flood test for upstream servers and reducing the size of the http2_integration_test.cc file

Risk Level: Low, test only
Testing: Unit Tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: None

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

**Comment by htuch** [comment]
Please merge master to pick the macOS CI fix.

File: test/integration/BUILD [hunk 1]
```diff
@@ -312,6 +312,30 @@ envoy_cc_test(
     ],
 )
 
+envoy_cc_test(
+    name = "http2_flood_integration_test",
+    srcs = [
+        "http2_flood_integration_test.cc",
+    ],
+    shard_count = 4,
```
Comment (asraa) [inline test/integration/BUILD:320]: shard_counts still needed?

**Review by asraa — COMMENTED** [review]
Thanks, LGTM

File: test/integration/BUILD (hunk 1 above)
Comment (yanavlasov) [reply test/integration/BUILD:320]: Yes, since we will be adding more tests for upstream flooding.

**Review by asraa — APPROVED** [review]
Thanks!

**Comment by yanavlasov** [comment]
Pending PR #13579

**Comment by yanavlasov** [comment]
Closing for a moment so it is merged before #13579

**Pull request merged**